### PR TITLE
Drop support for julia 1.0-1.3 

### PR DESCRIPTION
**Pull request by lgoettgens**
Resolves https://github.com/JuliaTesting/Aqua.jl/issues/216, closes https://github.com/JuliaTesting/Aqua.jl/pull/195.

### CODE REVIEW

**Comment by lgoettgens** [comment]
@fingolfin could you please mark the ci jobs for 1.0-1.3 as no longer required?

**Comment by hyrodium** [comment]
Usually, dropping support for earlier version of Julia is not considered as breaking.

>Changing Julia version compatibility must be a non-breaking feature
https://github.com/SciML/ColPrac#dropping-support-for-earlier-versions-of-julia

**Comment by lgoettgens** [comment]
> Usually, dropping support for earlier version of Julia is not considered as breaking.
> 
> > Changing Julia version compatibility must be a non-breaking feature
> 
> https://github.com/SciML/ColPrac#dropping-support-for-earlier-versions-of-julia

This is true for packages with a x.y.z versioning scheme. As Aqua.jl is on version 0.y.z, I can either release this as breaking, or as a patch release. But the julia version should not be restricted on a patch release.

**Comment by lgoettgens** [comment]
And there will be more breaking changes released together with this in one batch.

**Review by fingolfin — APPROVED** [review]

File: Project.toml [hunk 1]
```diff
@@ -1,18 +1,16 @@
 name = "Aqua"
 uuid = "4c88cf16-eb10-579e-8560-4a9242c79595"
 authors = ["Takafumi Arakaki <aka.tkf@gmail.com> and contributors"]
-version = "0.7.4"
+version = "0.8.0-DEV"
```
[the remark this answers is not in the document]
Comment (fingolfin) [reply Project.toml:4]: Nevermind, I see you already answered this in separate comments.

**Comment by fingolfin** [comment]
Oh, I'll also add the new 1.4 test for Windows and macOS to the list of required tests.

**Comment by lgoettgens** [comment]
> Oh, I'll also add the new 1.4 test for Windows and macOS to the list of required tests.

Can you add "enforce PR labels" as well?

**Comment by fingolfin** [comment]
@lgoettgens BTW you could have just restarted that one failing job, instead of restarting all :-)

**Comment by lgoettgens** [comment]
> @lgoettgens BTW you could have just restarted that one failing job, instead of restarting all :-)

I know, but I wanted to make sure that this is not something failing on a small percentage of runs, and more runs would maybe find that :)

**Comment by lgoettgens** [comment]
Windows-1.4 seems to need a bit more time than 10 minutes. I tried to lengthen the timeout for windows jobs.

**Pull request merged**